pennylane/operation.py

```diff
@@ -1761,10 +1765,43 @@ def matrix(self, wire_order=None):
             # append diagonalizing unitary for specific wire to U_list
             U_list.append(mats[0])
 
+        mat_size = np.prod([np.shape(mat)[0] for mat in U_list])
```

Comment:
Just double checking, but there is no case at the moment where `mat` will _not_ be a numpy array right?

Because tensor products atm only contain Pauli or Hadamard matrices, which take no parameters, so which will always be NumPy arrays

Reply:
I don't think there is any need to worry about this now, but later on, when we extend `@` to work with arbitrary operators, we will need to take this into account.

Reply:
As far as I can tell, the line 
```python
mats = [i.matrix() for i in g]
```
would already be problematic, because we are not providing any `op.data` to the matrix method there, right? But very good to have this on our minds, it will need adapting if we allow more observables with `@`...